crates/turbopack-ecmascript/src/references/esm/export.rs

```diff
@@ -216,6 +238,9 @@ impl CodeGenerateable for EsmExports {
                     body.insert(0, stmt);
                 }
             }
+            if let Some(cjs_stmt) = cjs_stmt.clone() {
+                insert_hoisted_stmt(program, cjs_stmt);
```

Comment:
It would be better if this statement is not inserted after *all* imports, but directly after the imported module.

This would behave more correct in circular dependencies

``` js
// entry.js
export * from "./commonjs.js";
import "./esm.js"

// esm.js
import { cjsExport } from "./entry.js"; // cycle

console.log(cjsExport);

// commonjs.js
exports.cjsExport = "Hello World";
```

Reply:
But this can be done in a later PR

Reply:
I'll postpone it for now